funcx_endpoint/funcx_endpoint/endpoint/endpoint.py

```diff
@@ -139,15 +139,20 @@ def configure_endpoint(
         )
 
     @staticmethod
-    def check_endpoint_json(endpoint_json, endpoint_uuid):
-        if os.path.exists(endpoint_json):
-            with open(endpoint_json) as fp:
-                log.debug("Connection info loaded from prior registration record")
-                reg_info = json.load(fp)
-                endpoint_uuid = reg_info["endpoint_id"]
-        elif not endpoint_uuid:
-            endpoint_uuid = str(uuid.uuid4())
-        return endpoint_uuid
+    def get_endpoint_id(endpoint_dir: pathlib.Path) -> str | None:
+        info_file_path = endpoint_dir / "endpoint.json"
+        try:
+            return json.loads(info_file_path.read_bytes())["endpoint_id"]
+        except FileNotFoundError:
+            pass
+        return None
+
+    @staticmethod
+    def check_endpoint_json(endpoint_dir: pathlib.Path, endpoint_uuid: str) -> str:
```

Comment:
You just moved get_endpoint_id right?  And refactored check_endpont_json to use it instead of duplicating logic.  It is cleaner with Path instead of joining the '/'.